### PR TITLE
Refactor/clang tidy cleanup

### PR DESCRIPTION
**Pull request by SteveBronder**
Edit: Something weird is going on with the Jenkinsbot so I move the branch over to the main repo

## Summary

Running clang-tidy with the checks below. This refactor is only aesthetic and does not change how any of the code is executed.

[readability-braces-around-statements](https://clang.llvm.org/extra/clang-tidy/checks/readability-braces-around-statements.html):
- Adds braces around conditionals and ifs

[modernize-use-using](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-using.html):
everywhere clang-tidy can replace a `typedef` with a `using` it will aka
`typedef double A;` becomes `using A = double;`

[modernize-use-nullptr](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-nullptr.html):
using `nullptr` instead of NULL

The original PR also had a cleanup to return a braced init list but after looking at the I didn't like it

## Tests

refactor is only aesthetic so no new tests

## Checklist

- [x] Math issue #1308

- [x] Copyright holder: Steve Bronder

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: stan/math/prim/scal/prob/beta_cdf.hpp [hunk 1]
```diff
@@ -37,8 +37,9 @@ return_type_t<T_y, T_scale_succ, T_scale_fail> beta_cdf(
   typedef partials_return_type_t<T_y, T_scale_succ, T_scale_fail>
       T_partials_return;
 
-  if (size_zero(y, alpha, beta))
+  if (size_zero(y, alpha, beta)) {
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/beta_cdf.hpp]: using at line 37

File: stan/math/prim/scal/prob/beta_lccdf.hpp [hunk 2]
```diff
@@ -42,8 +42,9 @@ return_type_t<T_y, T_scale_succ, T_scale_fail> beta_lccdf(
   typedef partials_return_type_t<T_y, T_scale_succ, T_scale_fail>
       T_partials_return;
 
-  if (size_zero(y, alpha, beta))
+  if (size_zero(y, alpha, beta)) {
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/beta_lccdf.hpp]: using at line 42

File: stan/math/prim/scal/prob/beta_lcdf.hpp [hunk 3]
```diff
@@ -42,8 +42,9 @@ return_type_t<T_y, T_scale_succ, T_scale_fail> beta_lcdf(
   typedef partials_return_type_t<T_y, T_scale_succ, T_scale_fail>
       T_partials_return;
 
-  if (size_zero(y, alpha, beta))
+  if (size_zero(y, alpha, beta)) {
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/beta_lcdf.hpp]: using at line 42

File: stan/math/prim/scal/prob/beta_lpdf.hpp [hunk 4]
```diff
@@ -55,10 +55,12 @@ return_type_t<T_y, T_scale_succ, T_scale_fail> beta_lpdf(
   check_nonnegative(function, "Random variable", y);
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/beta_lpdf.hpp]: using at line 46

File: stan/math/prim/scal/prob/exp_mod_normal_cdf.hpp [hunk 5]
```diff
@@ -25,8 +25,9 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_cdf(
                                           T_inv_scale>::type T_partials_return;
 
   T_partials_return cdf(1.0);
-  if (size_zero(y, mu, sigma, lambda))
+  if (size_zero(y, mu, sigma, lambda)) {
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/exp_mod_normal_cdf.hpp]: using at line 23

File: stan/math/prim/scal/prob/exp_mod_normal_lccdf.hpp [hunk 6]
```diff
@@ -24,8 +24,9 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_lccdf(
       T_partials_return;
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/exp_mod_normal_lccdf.hpp]: using line 23

File: stan/math/prim/scal/prob/exp_mod_normal_lcdf.hpp [hunk 7]
```diff
@@ -25,8 +25,9 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_lcdf(
                                           T_inv_scale>::type T_partials_return;
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/exp_mod_normal_lcdf.hpp]: using at line 23

File: stan/math/prim/scal/prob/exp_mod_normal_lpdf.hpp [hunk 8]
```diff
@@ -24,8 +24,9 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_lpdf(
   typedef partials_return_type_t<T_y, T_loc, T_scale, T_inv_scale>
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/exp_mod_normal_lpdf.hpp]: using line 23

File: stan/math/prim/scal/prob/pareto_type_2_cdf.hpp [hunk 9]
```diff
@@ -21,8 +21,9 @@ return_type_t<T_y, T_loc, T_scale, T_shape> pareto_type_2_cdf(
   typedef partials_return_type_t<T_y, T_loc, T_scale, T_shape>
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/pareto_type_2_cdf.hpp]: using

File: stan/math/prim/scal/prob/pareto_type_2_lccdf.hpp [hunk 10]
```diff
@@ -21,8 +21,9 @@ return_type_t<T_y, T_loc, T_scale, T_shape> pareto_type_2_lccdf(
   typedef partials_return_type_t<T_y, T_loc, T_scale, T_shape>
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/pareto_type_2_lccdf.hpp]: using

File: stan/math/prim/scal/prob/pareto_type_2_lcdf.hpp [hunk 11]
```diff
@@ -22,8 +22,9 @@ return_type_t<T_y, T_loc, T_scale, T_shape> pareto_type_2_lcdf(
   typedef partials_return_type_t<T_y, T_loc, T_scale, T_shape>
```
Comment (rok-cesnovar) [inline stan/math/prim/scal/prob/pareto_type_2_lcdf.hpp]: using

File: stan/math/rev/mat/fun/typedefs.hpp [hunk 12]
```diff
@@ -14,37 +14,37 @@ typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>::Index size_type;
  * The type of a matrix holding <code>var</code>
```
Comment (rok-cesnovar) [inline stan/math/rev/mat/fun/typedefs.hpp]: using at line 11

**Comment by SteveBronder** [comment]
@rok-cesnovar thanks for the review! I think I cleaned up all the rest of the typedefs. I left the ones in the MPI code that are used in a macro because I was worried about touching those. But everywhere else should be good to go

**Pull request merged**